manager/runner_test.go

```diff
@@ -942,6 +942,122 @@ func TestRunner_Start(t *testing.T) {
 		}
 	})
 
+	t.Run("multi-template-no-render-cycle", func(t *testing.T) {
+		testConsul.SetKVString(t, "multi-template-no-cycle-foo", "bar")
+
+		out1, err := os.CreateTemp("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.Remove(out1.Name())
+
+		out2, err := os.CreateTemp("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.Remove(out2.Name())
+
+		minWait := 1 * time.Second
+		maxWait := 10 * time.Second
+
+		c := config.DefaultConfig().Merge(&config.Config{
+			Wait: &config.WaitConfig{
+				Min: &minWait,
+				Max: &maxWait,
+			},
+			Consul: &config.ConsulConfig{
+				Address: config.String(testConsul.HTTPAddr),
+			},
+			Templates: &config.TemplateConfigs{
+				&config.TemplateConfig{
+					Contents:    config.String(`{{ key "multi-template-no-cycle-foo" }}`),
+					Destination: config.String(out1.Name()),
+				},
+				&config.TemplateConfig{
+					Contents:    config.String(`foobar`),
+					Destination: config.String(out2.Name()),
+				},
+			},
+		})
+		c.Finalize()
+
+		r, err := NewRunner(c, false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		go r.Start()
+		defer r.Stop()
+
+		// The template with no deps will be available first
```

Comment:
This is actually guaranteed.  When the template with the Consul key runs, it will [return early](https://github.com/hashicorp/consul-template/blob/main/manager/runner.go#L867-L879) because it hasn't created the Consul watcher yet.